channel-api/src/main/java/com/synopys/integration/alert/channel/api/convert/ProjectMessageConverter.java

```diff
@@ -0,0 +1,177 @@
+            chunkedStringBuilder.append(messageFormatter.getSectionSeparator());
+            chunkedStringBuilder.append(messageFormatter.getLineSeparator());
+        }
+
+        return chunkedStringBuilder.collectCurrentChunks();
+    }
+
+    private List<String> gatherBomComponentPieces(BomComponentDetails bomComponent) {
+        List<String> bomComponentSectionPieces = new LinkedList<>();
+
+        String componentString = createLinkableItemString(bomComponent.getComponent(), true);
+        bomComponentSectionPieces.add(componentString);
+        bomComponentSectionPieces.add(messageFormatter.getLineSeparator());
+
+        bomComponent.getComponentVersion()
+            .map(componentVersion -> createLinkableItemString(componentVersion, true))
+            .ifPresent(componentVersionString -> {
+                bomComponentSectionPieces.add(componentVersionString);
+                bomComponentSectionPieces.add(messageFormatter.getLineSeparator());
+            });
+
+        String nonBreakingSpace = messageFormatter.getNonBreakingSpace();
+
+        List<String> componentAttributeStrings = gatherAttributeStrings(bomComponent);
+        for (String attributeString : componentAttributeStrings) {
+            bomComponentSectionPieces.add(String.format("%s-%s%s", nonBreakingSpace, nonBreakingSpace, attributeString));
+            bomComponentSectionPieces.add(messageFormatter.getLineSeparator());
+        }
+
+        Set<ComponentConcern> sortedConcerns = new TreeSet<>(bomComponent.getComponentConcerns());
+        ComponentConcernType currentType = null;
+        ItemOperation currentOperation = null;
+        ComponentConcernSeverity currentSeverity = null;
+        for (ComponentConcern componentConcern : sortedConcerns) {
+            if (!componentConcern.getType().equals(currentType)) {
+                currentType = componentConcern.getType();
+                currentOperation = null;
+
+                bomComponentSectionPieces.add(messageFormatter.getLineSeparator());
+                bomComponentSectionPieces.add(messageFormatter.encode(currentType.name()));
+            }
+
+            if (!componentConcern.getOperation().equals(currentOperation)) {
+                currentOperation = componentConcern.getOperation();
+                currentSeverity = null;
+
+                bomComponentSectionPieces.add(messageFormatter.getLineSeparator());
+                bomComponentSectionPieces.add(nonBreakingSpace + nonBreakingSpace);
+                bomComponentSectionPieces.add(messageFormatter.encode(currentOperation.name()));
+            }
+
+            if (!componentConcern.getSeverity().equals(currentSeverity)) {
+                currentSeverity = componentConcern.getSeverity();
+
+                bomComponentSectionPieces.add(messageFormatter.getLineSeparator());
+                bomComponentSectionPieces.add(nonBreakingSpace + nonBreakingSpace + nonBreakingSpace + nonBreakingSpace);
```

Comment:
Is this an empty section with just 4 spaces? This part is a little confusing. Can we add a variable for spacings like this to maybe make it clearer why we're doing this?